### PR TITLE
Solves #1166

### PR DESCRIPTION
**Pull request by ColdanR**
Added encoding as parameter to set output file encoding

### CODE REVIEW

File: tools/maven-plugin/README.adoc [hunk 1]
```diff
@@ -49,6 +49,7 @@ E.g. for includeDependenciesScopes could be configured as:
 - `configProperties` (String) - Load any properties from a file. Example `${basedir}/src/main/resources/application.properties`.
 - `attachArtifacts` (boolean, default: false) - Attach the built OpenAPI schema as build artifact.
 - `skip` (boolean, default: false) - Skip execution of the plugin.
+- `encoding` (String) - Encoding of output OpenAPI files.
```
Comment (phillip-kruger) [inline tools/maven-plugin/README.adoc:52]: We can probably make the default UTF-8 to stay backward compatible ?

File: tools/maven-plugin/src/main/java/io/smallrye/openapi/mavenplugin/GenerateSchemaMojo.java [hunk 2]
```diff
@@ -223,6 +227,12 @@ public class GenerateSchemaMojo extends AbstractMojo {
     @Parameter(property = "scanExcludeProfiles")
     private List<String> scanExcludeProfiles;
 
+    /**
+     * Output encoding for openapi document.
+     */
+    @Parameter(property = "encoding")
```
Comment (phillip-kruger) [inline tools/maven-plugin/src/main/java/io/smallrye/openapi/mavenplugin/GenerateSchemaMojo.java:233]: So here defaultValue = "UTF-8"

**Review by phillip-kruger — CHANGES_REQUESTED** [review]

File: tools/maven-plugin/README.adoc (hunk 1 above)
Comment (MikeEdgar) [reply tools/maven-plugin/README.adoc:52]: @phillip-kruger, I thought the same thing, but previously it was using `String#getBytes`, which internally calls `Charset.defaultCharset`.

Comment (phillip-kruger) [reply tools/maven-plugin/README.adoc:52]: Ok, but it should still be able to do this. Not sure if you can make these properties Optional ? If so , use Optional with empty setting to default

Comment (MikeEdgar) [reply tools/maven-plugin/README.adoc:52]: The default `defaultValue` in the `@Parameter` is an empty string. That flows into the `(!StringUtils.isBlank(encoding))` check and then uses the platform default encoding.

Comment (phillip-kruger) [reply tools/maven-plugin/README.adoc:52]: Ok great !

Comment (phillip-kruger) [reply tools/maven-plugin/README.adoc:52]: So then we can merge ?

**Review by MikeEdgar — APPROVED** [review]

File: tools/maven-plugin/README.adoc (hunk 1 above)
Comment (MikeEdgar) [reply tools/maven-plugin/README.adoc:52]: Yeah, lgtm!

**Review by phillip-kruger — APPROVED** [review]
Thanks !

**Pull request merged**